### ISSUE TITLE
Error building with Pip

### ISSUE THREAD

**Issue by silascutler** [frida/frida#131]
Hello, When trying to install via pip, I get an error about os-version being referenced before assignment.  Seems to only happen on Windows. 

    running build_ext
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-IhDSEo/frida/setup.py", line 138, in <module>
        'build_ext': FridaPrebuiltExt
      File "/usr/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/python2.7/site-packages/setuptools/command/install.py", line 61                                                                                                 , in run
        return orig.install.run(self)
      File "/usr/lib/python2.7/distutils/command/install.py", line 563, in run
        self.run_command('build')
      File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/python2.7/distutils/command/build.py", line 127, in run
        self.run_command(cmd_name)
      File "/usr/lib/python2.7/distutils/cmd.py", line 326, in run_command
        self.distribution.run_command(command)
      File "/usr/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/usr/lib/python2.7/site-packages/setuptools/command/build_ext.py", line                                                                                                  50, in run
        _build_ext.run(self)
      File "/usr/lib/python2.7/distutils/command/build_ext.py", line 337, in run
        self.build_extensions()
      File "/usr/lib/python2.7/distutils/command/build_ext.py", line 446, in build_e                                                                                                 xtensions
        self.build_extension(ext)
      File "/tmp/pip-build-IhDSEo/frida/setup.py", line 60, in build_extension
        os_version=os_version
    UnboundLocalError: local variable 'os_version' referenced before assignment

**Comment by cganas** [comment]
How was this resolved?

**Comment by oleavr** [comment]
@cganas This was fixed with a [hack](https://github.com/frida/frida-python/blob/8ca9c0abb70bf5081c632a53cabc4b37cd075dc3/src/setup.py#L66-L95) where we detect that we're in a source package and pretend we build the native extension, but actually download the .egg and extract the prebuilt extension from it. It's pretty terrible, but haven't found a better option for now.

**Comment by cganas** [comment]
Ahh, okay. So is it still supposed to be happening? Trying to `pip install frida` on Windows 8.

**Comment by oleavr** [comment]
@cganas Oh, nice catch! Thanks! This is definitely a bug. So `target_extension` is `.dll` but the one in the egg is `.pyd`, correct? Sounds like we need a kludge for Windows, where we set `target_extension` to `.dll` regardless.

**Comment by cganas** [comment]
Yeah, hacking in `target_extension = ".pyd"` fixes it as far as I can tell.

I'm facing another issue with:
```
***
Failed to load the Frida native extension: Permission denied
Please ensure that the extension was compiled for Python 2.7.
***
```

However, I think that might be local to me. I'm going to continue poking around. Once I resolve it I can set up a pull request with the changes.

**Comment by oleavr** [comment]
@cganas Awesome!

**Comment by cganas** [comment]
I've been bashing my head over importing the actual native extension. I figured someone else might have more experience with it so I'll explain what's currently wrong:

I'm running on a 64bit system; however, I didn't realize my python is 32bit. So I let it pull the amd64 build, as expected, I get a format exec error when importing `_frida` but it did seem to find it.

However, if I overwrite that 64bit dll with the 32bit version I get "Failed to load the Frida native extension: No such file or directory" even though it is otherwise identical.

Tried renaming it to `.pyd` (even though other .dlls are successfully being imported) with no luck.
